mapbox/services/matrix.py

```diff
@@ -0,0 +1,41 @@
+"""Matrix API V1"""
+
+import re
+
+from mapbox.encoding import encode_waypoints
+from mapbox.errors import InvalidProfileError
+from mapbox.services.base import Service
+
+
+class DirectionsMatrix(Service):
+    """Access to the Matrix API V1"""
+
+    api_name = 'directions-matrix'
+    api_version = 'v1'
+
+    valid_profiles = [
+        'mapbox/driving', 'mapbox/cycling', 'mapbox/walking',
+        'mapbox/driving-traffic']
+
+    @property
+    def baseuri(self):
+        return 'https://{0}/{1}/{2}'.format(
+            self.host, self.api_name, self.api_version)
+
+    def _validate_profile(self, profile):
+        parts = re.split(r'[\/\.]', profile)
+        if len(parts) == 1:
+            parts = ['mapbox'] + parts
+        profile = '/'.join(parts)
+        if profile not in self.valid_profiles:
+            raise InvalidProfileError(
+                "{0} is not a valid profile".format(profile))
+        return profile
+
+    def matrix(self, features, profile='mapbox/driving'):
```

Comment:
This class has only one method. We could have used `get()` or even `__call__()` but I prefer a more significant name.

Reply:
This is the case with all of our service classes - agree that the explicit name is better but _if_ we wanted to standardize the method name or make the class callable, we should do so across the board before the 1.0 release.